recipe_scrapers/weightwatchers.py

```diff
@@ -0,0 +1,152 @@
+# mypy: allow-untyped-defs
+
+import re
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, get_yields, normalize_string
+
+
+class Weightwatchers(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "www.weightwatchers.com"
+
+    def author(self):
+        return "WeightWatchers"
+
+    def title(self):
+        return self.soup.find("h1").get_text().strip()
+
+    def category(self):
+        return "WeightWatchers"
+
+    # cooking times, yield, difficulty are in a common div in public and non-public recipes
+    # but class of that block and sub elements are different
+    # so finding the block and extracting a value will be overridden in class for public recipes,
+    # but picking the data item based on order is don in this base class (total_time(), cook_time() and so on)
+    def _findDataContainer(self):
+        return self.soup.find("div", {"class": "styles_container__3N3E8"})
+
+    def _extractItemField(self, item):
+        return item.contents[1]
+
+    def total_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[0])
+        )
+
+    def prep_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[1])
+        )
+
+    def cook_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[2])
+        )
+
+    def yields(self):
+        return get_yields(self._extractItemField(self._findDataContainer().contents[3]))
+
+    def difficulty(self):
+        return self._extractItemField(self._findDataContainer().contents[4]).get_text()
+
+    #   Alternative way to extract data based on description instead of position
+    #    def total_time(self):
+    #        return get_minutes(
+    #            self.__findDataContainer()
+    #            .find("div", string=re.compile(r"minutes Total Time"))
+    #            .previous_sibling
+    #        )
+
+    def image(self):
+        backgroundImgStyle = self.soup.find("div", {"class": "styles_image__2dnNm"})[
+            "style"
+        ]
+
+        if backgroundImgStyle:
+            return (
+                re.compile(r'url\("(?P<imgurl>\S*)"\);')
```

Comment:
nit: `compile()` can be removed since we're not saving the object out of it. 

More information: https://stackoverflow.com/a/47269110